mobile/mobile.go

```diff
@@ -1,33 +1,34 @@
 package mobile
 
 import (
-	"encoding/json"
-
 	mh "gx/ipfs/QmPnFwZ2JXKnXgMw8CdBPxn7FWh6LLdjUjxV1fKHuJnkr8/go-multihash"
 	logging "gx/ipfs/QmZChCsSt8DctjceaL56Eibc29CVQq4dGKRXC5JRZ6Ppae/go-log"
 
+	"github.com/golang/protobuf/proto"
 	"github.com/textileio/textile-go/broadcast"
 	"github.com/textileio/textile-go/core"
 	"github.com/textileio/textile-go/keypair"
+	"github.com/textileio/textile-go/pb"
 	"github.com/textileio/textile-go/wallet"
 )
 
 var log = logging.Logger("tex-mobile")
 
-// Message is a generic go -> bridge message structure
-type Event struct {
-	Name    string `json:"name"`
-	Payload string `json:"payload"`
-}
-
-// Messenger is used to inform the bridge layer of new data waiting to be queried
+// Messenger is a push mechanism to the bridge
 type Messenger interface {
 	Notify(event *Event)
 }
 
-// Callback is used for asyc methods (payload is a protobuf)
+// Event is sent by Messenger to the bridge (data is a protobuf,
+// name is the string value of a pb.MobileEvent_Type)
+type Event struct {
+	Name string
+	Data []byte
```

Comment:
mobile's `Event` now delivers a byte slice, which is always a marshaled proto message.